pkg/resources/grant_ownership.go

```diff
@@ -0,0 +1,668 @@
+				onAllOrFuture["in_schema"] = data.Schema.FullyQualifiedName()
+			}
+
+			switch id.Kind {
+			case OnAllGrantOwnershipKind:
+				on["all"] = []any{onAllOrFuture}
+			case OnFutureGrantOwnershipKind:
+				on["future"] = []any{onAllOrFuture}
+			}
+
+			if err := d.Set("on", []any{on}); err != nil {
+				return nil, err
+			}
+		}
+
+		return []*schema.ResourceData{d}, nil
+	}
+}
+
+func CreateGrantOwnership(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id, err := createGrantOwnershipIdFromSchema(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	logging.DebugLogger.Printf("[DEBUG] created identifier from schema: %s", id.String())
+
+	grantOn, err := getOwnershipGrantOn(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = client.Grants.GrantOwnership(
+		ctx,
+		grantOn,
+		getOwnershipGrantTo(d),
+		getOwnershipGrantOpts(id),
+	)
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "An error occurred during grant ownership",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", id.String(), err),
+			},
+		}
+	}
+
+	logging.DebugLogger.Printf("[DEBUG] Setting identifier to %s", id.String())
+	d.SetId(id.String())
+
+	return ReadGrantOwnership(ctx, d, meta)
+}
+
+func DeleteGrantOwnership(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id, err := ParseGrantOwnershipId(d.Id())
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to parse internal identifier",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+			},
+		}
+	}
+
+	grantOn, err := getOwnershipGrantOn(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if grantOn.Future != nil {
+		// TODO (SNOW-1182623): Still waiting for the response on the behavior/SQL syntax we should use here
+	} else {
+		accountRoleName, err := client.ContextFunctions.CurrentRole(ctx)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		err = client.Grants.GrantOwnership( // TODO: Should we always set outbound privileges to COPY in delete operation or set it to the config value?
+			ctx,
+			grantOn,
+			sdk.OwnershipGrantTo{
+				AccountRoleName: sdk.Pointer(sdk.NewAccountObjectIdentifier(accountRoleName)), // TODO: What if current role is database role (not a common but possible case)
+				// DatabaseRoleName: TODO: handle later
+			},
+			getOwnershipGrantOpts(id),
+		)
+		if err != nil {
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  "An error occurred when transferring ownership back to the original role",
+					Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+				},
+			}
+		}
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func ReadGrantOwnership(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
```

Comment:
so are there cases where we should setId to `""` in read like in the normal resources?

Reply:
As discussed, every time where we cannot find the ownership on the target object